### ISSUE TITLE
Date Palms missing trunk nodes on sapling spawn.

### ISSUE THREAD

**Issue by daytonnvpaul** [mt-mods/moretrees#6]
Date Palms missing trunk nodes revisited.

Verified once again on my local server. Trunks are missing from y position 14 through 17. The woodcutter mod stops cutting at the missing trunk.

I have yet to spawn a Date Palm, and had these 4 trunks appear in those positions.

(Notice that I removed the reference to the technic chainsaw, as I don't have that mod installed on this server.)

**Comment by S-S-X** [comment]
I've no knowledge about this issue but I guess screenshots could be helpful.

**Comment by daytonnvpaul** [comment]
I find it hard to understand how you can't simply spawn one on a local game.  It's quite obvious that 4 tree trunks are missing from position 14 through 17.  that would be 14 nodes above ground level, or where the sapling was placed.

Sorry if this seems infantile, but it seems if I call the issue, and simply mention that I didn't know if the technic chainsaw had the same efefect, many decided to test with the chainsaw, rather than address the issue of "MISSING TRUNK NODES".

If they can test with technic, I'm quite sure they could have checked for the initial issue of missing trunks.

Once again, please note that I do not have technic on this particular installation, so I don't really care if the technic chainsaw takes the whole tree.

As for linking to #1, it was clearly noted to open a new issue.

**Comment by S-S-X** [comment]
No need to go into aggressively defensive mode, I'd suggest cooperative mode instead.
Screenshot request was to help understand issue without having to launch game, there's other ways to handle people who'd be editing screenshots for bug reports, no need to discuss that here...
Always link if there's similar issue, especially if there was request to open new issue.

**Comment by wsor4035** [comment]
this uses minetest l system to define the tree found here https://github.com/mt-mods/moretrees/blob/master/tree_models.lua#L143-L166 As you can see in the attached issue, is that there is one singular tree trunk node up in the leaves, which means tools that crawl up the trunk arent going to go that far (also op numbers where slightly off from my examples).

![image](https://user-images.githubusercontent.com/24964441/170836932-e89003a5-c84a-4189-b2cf-885d5ca601a0.png)

**Comment by S-S-X** [comment]
So reason why technic chainsaw works is simply because it also traverses leaves, not just trunk.
Could also ask @SwissalpS does this look like your trees?
Seem like problem to me anyway, not very big problem but still something that should be fixed.

**Comment by daytonnvpaul** [comment]
S-S-X, this is the reason I went into an "aggressively defensive" mode.  I am NOT using technic chainsaw, so that explanation or discussion is moot.

wsor4035, your screenshot shows the gap between the 14th and 17th tree trunk, so it proves my description of the missing tree trunks.

Now, when using the woodcutter mod, it stops cutting before reaching the gap in the tree, leaving an ugly floating partial tree.

Please note, once again, I am not using technic chainsaw.  Please don't reference that to the issue I've posted.

As for referring to my other issue, I would have to say that with only 4 issues on this mod, it really shouldn't be necessary to find my original post. :)

**Comment by daytonnvpaul** [comment]
As requested, here are 2 screenshots.  The first is the Date Palm tree as spawned using bonemeal on the sapling.  The second is after using WorldEdit to replace the leaves with air.  I believe this shows the location of the missing tree trunks.

![screenshot_20220528_132955](https://user-images.githubusercontent.com/23373456/170842000-c5f0e723-b976-4575-a5fb-802767c2bd93.png)


![screenshot_20220528_133219](https://user-images.githubusercontent.com/23373456/170842010-377bc5bc-685e-45ac-bafb-e5b757d89608.png)

**Comment by SwissalpS** [comment]
> Could also ask @SwissalpS does this look like your trees? Seem like problem to me anyway, not very big problem but still something that should be fixed.
A date palm planted years ago on pandorabox:
![datePalmOld2022-05-31 19-44-16](https://user-images.githubusercontent.com/161979/171252724-ec52be85-bca6-4c0b-8aff-3859435c2636.png)
One planted today, also on pandorabox:
![datePalmNew2022-05-31 19-48-59](https://user-images.githubusercontent.com/161979/171252810-cb8308f0-33b7-4408-b91d-492a1b6173a3.png)

I used woodcutting mod to cut it down, and yes I can confirm that it doesn't continue all the way up.
What if woodcutting mod would also follow leaves of the tree that the trunk belongs to (or any for that matter)?
As that seems to be the problem. I have no problem with the trees looking this way.
Adding trunks will change the trunk/leaves balance - probably not going to hurt many games.
But the actual problem imo, is that woodcutting mod leaves the leaves and thus sometimes a mess, letting them drop.

In short: I consider this a woodcutting 'bug' not a moretrees issue.

**Comment by daytonnvpaul** [comment]
I also have no problem with the tree looking like this, but it seems as if even 1 more trunk node would solve the woodcutter failing to cut the entire tree.

Also, I don't think its very fair to put the blame on the woodcutter mod, as the Date Palm is the only tree that fails.  I have enabled every species from moretrees, and woodcutter removes all but the Date Palm.   The same with cooltrees mod.  Woodcutter removes all of them.

If I understood the template used for the Date Palm, I would have made the changes locally, and not posted the issue.

**Comment by SwissalpS** [comment]
Tree designers should not be held accountable for other mods that happen to work on trees.
I feel they should be free to design their trees whichever way they feel is good.
Especially since it is possible to design the cut-down mods in a way that it works - as demonstrated with the technic chainsaw.

**Comment by daytonnvpaul** [comment]
In that case, assuming you are on the path to closing this, how can I learn how to edit my own copy of the tree, so I can continue to use the woodcutter mod as is.

From my point of view, if only one tree doesn't work properly with woodcutter, then the issue isn't with woodcutter.

And again, please stop using technic as the baseline, as I don't use technic, and it seems as if many other admins have moved away from technic in the last few years.

**Comment by OgelGames** [comment]
No, @daytonnvpaul is right, the tree clearly is the problem here, have you ever seen a tree in real life with part of its trunk floating, unconnected to the rest of its trunk?

The code that defines the trunk portion of the tree is here:
https://github.com/mt-mods/moretrees/blob/62cab1b1d9d40a73b2f304650951d4659d86e391/tree_models.lua#L149

The section where the trunks are missing is the `[TGGGGT]` at the end, trunks where the `T`'s are, and empty space where the `G`'s are.

I think the reason behind this is to make it look like the trunk ends when the leaves start, while still having a trunk node within range of the topmost leaves to prevent decaying. Back when date palms were added to moretrees (2016), leaf decay was still ABM based, so there always needed to be a trunk node within range, otherwise the leaves would decay even without a player touching them.

**Comment by daytonnvpaul** [comment]
@OgelGames Thank you.

**Comment by SwissalpS** [comment]
> have you ever seen a tree in real life with part of its trunk floating, unconnected to the rest of its trunk?

no, but I can imagine that the original artist wanted to ilustrate the big bulb date palms have at the top without adding even more trunk nodes.
